morebits.js

```diff
@@ -1308,29 +1308,238 @@ Morebits.unbinder.getCallback = function UnbinderGetCallback(self) {
 
 
 /**
- * **************** Date ****************
- * Helper functions to get the month as a string instead of a number
- *
- * Normally it is poor form to play with prototypes of primitive types, but it
- * is fairly unlikely that anyone will iterate over a Date object.
+ * **************** Morebits.date ****************
+ */
+
+/**
+ * @constructor
+ * Create a date object. MediaWiki timestamp format is also acceptable,
+ * in addition to everything that JS Date() accepts.
  */
+Morebits.date = function() {
+	var args = Array.prototype.slice.call(arguments);
 
-Date.monthNames = ['January', 'February', 'March', 'April', 'May', 'June',
-	'July', 'August', 'September', 'October', 'November', 'December' ];
+	this._d = new (Function.prototype.bind.apply(Date, [Date].concat(args)));
```

Comment:
Basically this line means `new Date.apply(args)` but it doesn't work that simply because of peculiarities of `Date`. In ES5, it isn't possible to create a class that inherits from `Date`. Also, for whatever reason, JS runtime engines have some low-level code to prevent date methods (`getDate` etc) from being applied to anything other than a `Date` object. So we must use an underlying `Date` object in the class. You can read up on all that by googling "how to extend the date class in ES5".

Reply:
Thanks!

>googling "how to extend the date class in ES5"

😆